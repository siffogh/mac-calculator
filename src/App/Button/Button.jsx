import React from 'react';
import PropTypes from 'prop-types';

import { getDataByValue } from '../../utils';

class Button extends React.Component {
  static propTypes = {
    value: PropTypes.string,
    span: PropTypes.string,
    nonInitialLabel: PropTypes.string
  };

  static defaultProps = {
    value: undefined,
    span: undefined,
    nonInitialLabel: undefined
  };

  static contextTypes = {
    initial: PropTypes.bool,
    handleInput: PropTypes.func
  };

  handleClick = () => {
    this.context.handleInput({ value: this.props.value });
  };

  render() {
    const { value = '', span = 1, nonInitialLabel } = this.props;
    const { initial } = this.context;
    const buttonData = getDataByValue({ value });

    if (!buttonData) {
      return <input type="button" />;
    }

    const { type, label } = buttonData;

    return (
      <input
        type="button"
        key={value}
        className={`${type} ${type}-${value}`}
        style={{ gridColumn: `span ${span}` }}
        onClick={this.handleClick}
        value={!initial && nonInitialLabel ? nonInitialLabel : label}
      />
    );
  }
}


export default Button;
