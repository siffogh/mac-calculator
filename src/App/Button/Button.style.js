import styled from 'styled-components';
import { font } from '../App.style';

const Input = styled.input`
  border: 1px solid #808080;
  outline: none;
`;

const Digit = Input.extend`
  ${font({ size: '6em' })} background: #e0e0e0;
  font-weight: 200;
`;

const Dot = Digit;

const Binary = Input.extend`
  ${font({ size: '5em' })} color: white;
  background: #f5923e;
`;

const Equal = Binary;

const Unary = Input.extend`
  ${font({ size: '5em' })} background: #d6d6d6;
  font-weight: 200;
`;

const Reset = Unary;

export default { Input, Digit, Dot, Binary, Unary, Equal, Reset };
