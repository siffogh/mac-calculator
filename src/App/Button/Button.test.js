import React from 'react';
import { mount } from 'enzyme';
import Button from './Button';

describe('button', () => {
  test('should render empty Button', () => {
    const wrapper = mount(<Button />);
    expect(wrapper.find('input[type="button"]')).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render empty button if value doesn\'t match', () => {
    const wrapper = mount(<Button value="somevalue" />);
    expect(wrapper.find('input[type="button"]')).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
