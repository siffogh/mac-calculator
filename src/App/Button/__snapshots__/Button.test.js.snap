// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should render empty Button 1`] = `
<Button>
  <input
    type="button"
  />
</Button>
`;

exports[`button should render empty button if value doesn't match 1`] = `
<Button
  value="somevalue"
>
  <input
    type="button"
  />
</Button>
`;
